### ISSUE TITLE
Unable to enter password when run a sample test case 

### ISSUE THREAD

**Issue by badarmunir** [Microsoft/EasyRepro#808]
Steps to reproduce
open easyrepro latest version in visual studio
build the solution
go to uci>create>CreateCase.cs run following test case

**EasyRepro Version**

  - [ ] Microsoft Dynamics 365 Online Version 9.1 (9.1.x) (DB 9.1.x) online
  
**UCI or Classic Web**
  - [ ] UCI (XrmApp)

**Online or On Premise**
  - [ ] Online

**Browser**
  - [ ] Chrome

**Describe the bug**  

**Code to reproduce**  

 
**Expected behavior**  
- Automation script should be able to login successfully

**Scr
![Capture](https://user-images.githubusercontent.com/23335796/76867828-e0d2b380-6887-11ea-9be0-4d7feb8e427e.PNG)
![issue](https://user-images.githubusercontent.com/23335796/76867837-e4663a80-6887-11ea-8463-54e9eb298674.PNG)

eenshots**  

**Additional context**

**Comment by badarmunir** [comment]
@TYLEROL kindly look into this issue

**Comment by gabyMe** [comment]
I had the same issue. The solution is to add a think time in the WebClient class ->  login method , before the EnterPassword statement.
But this means that you need to modify the framework and is not a good practice.
Also, regarding the MFA login, in Online, I discovered that we were using another Element reference, for the verification code, so I needed to modify also the "Login_OneTimeCode" from "//input[@name='otc']" to "//input[@name='VerificatioCode']". Maybe it will be the case in your situation also.
In conclusion, if you want to use the login with the MFA, you need to do your own modification in the framework. At least this is how I resolved the issue.

**Comment by TYLEROL** [comment]
@badarmunir - Is this still occurring? It looks like the failure is likely occurring around Line 124 of WebClient.cs (the EnterUsername() method. If this returns false, then you get down into a path of code that eventually leads to your error.

```
            bool success = EnterUserName(driver, username);
            if (!success)
            {
                var isUserAlreadyLogged = IsUserAlreadyLogged();
                if (isUserAlreadyLogged)
                {
                    SwitchToDefaultContent(driver);
                    return LoginResult.Success;
                }

                ThinkTime(1000);
                waitingForOtc = GetOtcInput(driver) != null;

                if (!waitingForOtc)
                    throw new Exception($"Login page failed. {Reference.Login.UserId} not found.");
            }
```

What login experience do you see when you test this? Is it a standard O365 cloud identity login? Or a federated experience? 

I'm going to make an assumption and assume this isn't federated as you have not provided a redirect action in your login method. If you are federated, then you may need to add in a redirect action to the login method in the sample.

In a scenario where you are not federated -- there is one possibility in which a 'Use Another Account' dialog appears. That should be handled by line 121 -- but important to observe if this appears and that it goes away to expose the Login_UserId element.

 If you see the Username field on the page, then it should be entering this portion of code:

```
        private bool EnterUserName(IWebDriver driver, SecureString username)
        {
            var input = driver.WaitUntilAvailable(By.XPath(Elements.Xpath[Reference.Login.UserId]), new TimeSpan(0, 0, 30));
            if (input == null)
                return false;

            input.SendKeys(username.ToUnsecureString());
            input.SendKeys(Keys.Enter);
            return true;
        }
```
The Login.UserId xpath should be looking for an **input** tag with **type** email. You can use the browser F12 dev tools Inspector to look at the page elements for the Username field (shown below).

`{ "Login_UserId", "//input[@type='email']"},`

Please let me know if this helps uncover what may be going wrong in the login sequence.

Warm Regards,
Tyler
